### PR TITLE
cloudcore support configmap dynamic update when the env of container …

### PR DESCRIPTION
**Pull request by rzyeleven**
…inject from configmap or secret

**What type of PR is this?**

/kind bug

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

fix issue [https://github.com/kubeedge/kubeedge/issues/2930](https://github.com/kubeedge/kubeedge/issues/2930)

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

**Review by fisherxu — COMMENTED** [review]
@rzyeleven Thanks for the fixing, and please fix the DCO and gofmt ci failed issue :)

File: cloud/pkg/edgecontroller/manager/location.go [hunk 1]
```diff
@@ -40,6 +40,17 @@ func (lc *LocationCache) PodConfigMapsAndSecrets(pod v1.Pod) (configMaps, secret
 				secrets = append(secrets, ef.SecretRef.Name)
 			}
 		}
+		for _,e := range s.Env {
```
Comment (fisherxu) [inline cloud/pkg/edgecontroller/manager/location.go:43]:
```suggestion
		for _, e := range s.Env {
```

**Comment by fisherxu** [comment]
image

Looks like github can't identify your github account, you need to add the email in the commit to your profile, Settings -> Emails -> Add email address. @rzyeleven

**Comment by rzyeleven** [comment]
> image
> 
> Looks like github can't identify your github account, you need to add the email in the commit to your profile, Settings -> Emails -> Add email address. @rzyeleven

thks, I've add an verify my email.

File: cloud/pkg/edgecontroller/manager/location.go [hunk 2]
```diff
@@ -40,6 +40,17 @@ func (lc *LocationCache) PodConfigMapsAndSecrets(pod v1.Pod) (configMaps, secret
...
 		}
+		for _, e := range s.Env {
+			if e.ValueFrom == nil {
+				continue
+			}
+
+			if e.ValueFrom.ConfigMapKeyRef != nil {
+				configMaps = append(configMaps, e.ValueFrom.ConfigMapKeyRef.Name)
+			} else if  e.ValueFrom.SecretKeyRef != nil {
+				secrets = append(configMaps, e.ValueFrom.SecretKeyRef.Name)
```
Comment (fisherxu) [inline cloud/pkg/edgecontroller/manager/location.go:51]:
```suggestion
				secrets = append(secrets, e.ValueFrom.SecretKeyRef.Name)
```

File: cloud/pkg/edgecontroller/manager/location.go [hunk 3]
```diff
@@ -40,6 +40,17 @@ func (lc *LocationCache) PodConfigMapsAndSecrets(pod v1.Pod) (configMaps, secret
...
 			}
 		}
+		for _, e := range s.Env {
+			if e.ValueFrom == nil {
+				continue
+			}
+
+			if e.ValueFrom.ConfigMapKeyRef != nil {
+				configMaps = append(configMaps, e.ValueFrom.ConfigMapKeyRef.Name)
+			} else if  e.ValueFrom.SecretKeyRef != nil {
```
Comment (fisherxu) [inline cloud/pkg/edgecontroller/manager/location.go:50]:
```suggestion
			} else if e.ValueFrom.SecretKeyRef != nil {
```

**Review by fisherxu — COMMENTED** [review]
/lgtm
/approve

**Pull request merged**